### ISSUE TITLE
Turn off class name compression by default and allow to opt-in

### ISSUE THREAD

**Issue by kof** [cssinjs/jss#1072]
Majority of users seem to value more the easer debuggability in production mode with readable class names than the download size optimization.

Should we turn it off by default and allow to opt-in, like [MUI](https://github.com/mui-org/material-ui/pull/15044) did?

**Comment by HenriBeck** [comment]
We could do it, but we should then include a prop on the jss provider to automatically do this

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kof** [comment]
Yes, react-jss provider and jss instance both need to have a new option for this.

**Comment by HenriBeck** [comment]
I think we could do that, but I would rather have 2 different class name generators in the core instead of one being configured by an option

**Comment by kof** [comment]
How would a user turn on compression?

**Comment by HenriBeck** [comment]
In react-jss I would provide a prop, but if the user decides to use compression he can just pass the other class name generator

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kof** [comment]
They would have to reimplement our generateId function by themselves, I fear this is too much for the most.

**Comment by HenriBeck** [comment]
No, people who use jss directly would just need to import a different createGenerateClassName

**Comment by kof** [comment]
So JSS would provide 2 generators, one with compression, another one without? I dunno...

**Comment by kof** [comment]
Or do you mean option for compression would exist in createGenerateId and user can create a generator with that option and set it in JSS?

**Comment by HenriBeck** [comment]
We could also add an options object to createGenerateClassName function which has a property called minify

**Comment by kof** [comment]
in that case what's the difference to having that option in Jss options?

**Comment by HenriBeck** [comment]
```js
const jss = create({
  classNameOptions: { minify: true },
});
```

```js
const createGenerateClassName = (options: { minify: boolean }) => {
  // code
};
```

This way, MUI would also only need one class name generator and could just use the options

**Comment by kof** [comment]
yeah, thats the original thought and an equivalent option can be provided through JssProvider as a prop. So, back to the origins.

**Comment by kof** [comment]
Since we renamed the generator to id, I would call it `{idOptions: {minify: true}` or even `{id: {minify: true}` since we are kinda in options object, we can consider everything being an option semantically.

**Comment by HenriBeck** [comment]
Merged in #1075
